Dockerfile

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright 2020 (C) Olliver Schinagl <oliver@schinagl.nl>
+
+# hadolint ignore=DL3007  latest is the latest stable for alpine
+FROM registry.hub.docker.com/library/alpine:latest AS builder
+
+LABEL Maintainer="Olliver Schinagl <oliver@schinagl.nl>"
+
+WORKDIR /src
+
+COPY . /src/
+
+# hadolint ignore=DL3008  We want the latest stable versions
+RUN apk add --no-cache \
+        bsd-compat-headers \
+        bzip2-dev \
+        cmake \
+        curl-dev \
+        file \
+        fts-dev \
+        g++ \
+        git \
+        json-c-dev \
+        libmilter-dev \
+        libtool \
+        libxml2-dev \
+        linux-headers \
+        make \
+        ncurses-dev \
+        openssl-dev \
+        pcre2-dev \
+        python3 \
+        zlib-dev \
+    && \
+    mkdir -p "./build" && \
+    cd "./build" && \
+    cmake .. \
+          -DAPP_CONFIG_DIRECTORY="/etc/clamav" \
+          -DCMAKE_BUILD_TYPE"=release" \
+          -DCMAKE_INSTALL_FULL_BINDIR="/usr/bin" \
+          -DCMAKE_INSTALL_FULL_INCLUDEDIR="/usr/include" \
+          -DCMAKE_INSTALL_FULL_LIBDIR="/usr/lib" \
+          -DCMAKE_INSTALL_FULL_SBINDIR="/usr/sbin" \
+          -DCMAKE_INSTALL_LIBDIR="/usr/lib" \
+          -DCMAKE_INSTALL_PREFIX="/usr" \
+          -DENABLE_CLAMONACC=OFF \
+          -DENABLE_DOCS=OFF \
+          -DENABLE_EXAMPLES=OFF \
+          -DENABLE_JSON_SHARED=ON \
+          -DENABLE_MILTER=ON \
+          -DENABLE_STATIC_LIB=OFF && \
+    make DESTDIR="/clamav" -j$(($(nproc) - 1)) install && \
+    rm -r \
+       "/clamav/usr/include" \
+       "/clamav/usr/lib/pkgconfig/" \
+    && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/freshclam.pid|" \
+        -e "s|.*\(DatabaseOwner\) .*|\1 clamav|" \
+        -e "s|^\#\(UpdateLogFile\) .*|\1 /var/log/clamav/freshclam.log|" \
+        -e "s|^\#\(NotifyClamd\).*|\1 /etc/clamav/clamd.conf|" \
+        -e "s|^\#\(ScriptedUpdates\).*|\1 yes|" \
+        -e "s|^\#\(AllowSupplementaryGroups\).*|\1 yes|" \
+        "/clamav/etc/clamav/freshclam.conf.sample" > "/clamav/etc/clamav/freshclam.conf" && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/clamd.pid|" \
+        -e "s|.*\(LocalSocket\) .*|\1 /run/clamav/clamd.sock|" \
+        -e "s|.*\(TCPSocket\) .*|\1 3310|" \
+        -e "s|.*\(TCPAddr\) .*|\1 0.0.0.0|" \
+        -e "s|.*\(User\) .*|\1 clamav|" \
+        -e "s|^\#\(LogFile\) .*|\1 /var/log/clamav/clamd.log|" \
+        -e "s|^\#\(LogTime\).*|\1 yes|" \
+        -e "s|^\#\(AllowSupplementaryGroups\).*|\1 yes|" \
+        "/clamav/etc/clamav/clamd.conf.sample" > "/clamav/etc/clamav/clamd.conf" && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/clamav-milter.pid|" \
+        -e "s|.*\(MilterSocket\) .*|\1 inet:7357|" \
+        -e "s|.*\(User\) .*|\1 clamav|" \
```

Comment:
It might be good to add these lines to match the settings above:
```
        -e "s|^\#\(LogFile\) .*|\1 /var/log/clamav/milter.log|" \
        -e "s|^\#\(LogTime\).*|\1 yes|" \
```